agent/recorder.go

```diff
@@ -140,38 +140,57 @@ func (r *SpanRecorder) loop() error {
 // Sends the spans in the buffer to Scope
 func (r *SpanRecorder) sendSpans() (error, bool) {
 	atomic.AddInt64(&r.stats.sendSpansCalls, 1)
-
 	spans := r.popSpans()
-	payload := r.getPayload(spans, r.metadata)
 
-	buf, err := encodePayload(payload)
-	if err != nil {
-		atomic.AddInt64(&r.stats.sendSpansKo, 1)
-		atomic.AddInt64(&r.stats.spansNotSent, int64(len(spans)))
-		return err, false
-	}
+	const batchSize = 1000
+	batchLength := len(spans) / batchSize
 
-	var testSpans int64
-	for _, span := range spans {
-		if isTestSpan(span) {
-			testSpans++
+	r.logger.Printf("sending %d spans in %d batches", len(spans), batchLength+1)
+
+	var lastError error
+	for b := 0; b <= batchLength; b++ {
+		var batch []tracer.RawSpan
+		if b == batchLength {
+			batch = spans[b*batchSize:]
```

Comment:
I'd add some comments to explain what it is doing that if you don't know the Collections manipulation in `Go`.

Reply:
ok